impl/src/main/java/com/okta/jwt/impl/jjwt/RemoteJwkSigningKeyResolver.java

```diff
@@ -42,7 +43,8 @@
     private final URL jwkUri;
     private final HttpClient httpClient;
     private final ObjectMapper objectMapper = new ObjectMapper();
-    private final Map<String, Key> keyMap = new HashMap<>();
+    private final Object lock = new Object();
+    private Map<String, Key> keyMap = new HashMap<>();
```

Comment:
It looks like `keyMap` should be `volatile` given the new structure of this code.

If this field gets read/written to from different threads, it might not have the most up to date value.

Reply:
Good call!